src/diffpy/srmise/applications/extract.py

```diff
@@ -419,82 +419,85 @@ def main():
     from diffpy.srmise.pdfpeakextraction import PDFPeakExtraction
     from diffpy.srmise.srmiseerrors import SrMiseDataFormatError, SrMiseFileError
 
-    try:
-        options.peakfunction = eval("peaks." + options.peakfunction)
-    except Exception as err:
-        print(err)
-        print("Could not create peak function '%s'. Exiting." % options.peakfunction)
-        return
-
-    try:
-        options.modelevaluator = eval("modelevaluators." + options.modelevaluator)
-    except Exception as err:
-        print(err)
-        print("Could not find ModelEvaluator '%s'. Exiting." % options.modelevaluator)
-        return
-
-    if options.bcrystal is not None:
+    if options.peakfunction:
+        try:
+            options.peakfunction = eval("peaks." + options.peakfunction)
+        except Exception as err:
+            print(err)
+            print("Could not create peak function '%s'. Exiting." % options.peakfunction)
```

Comment:
These error messages could be improved by giving the user hints about how to fix them.  Let's make an issue to make them more informative for the user, but don't worry about closing that issue in this release.

In general we need tests in srmise (please can you make an issue for that too?)

Reply:
issue created